detection-rules/body_business_email_compromise_new_sender.yml

```
@@ -15,11 +15,34 @@ source: |
       // out of office auto-reply
       // the NLU model will handle these better natively soon
       or strings.istarts_with(subject.subject, "Automatic reply:")
-      or regex.imatch(subject.subject, '(\[[^\]]+\]\s?){0,3}(re|fwd?|automat.*)\s?:.*')
+      or regex.imatch(subject.subject,
+                      '(\[[^\]]+\]\s?){0,3}(re|fwd?|automat.*)\s?:.*'
+      )
     )
     and (
-      length(headers.references) > 0
-      or any(headers.hops, any(.fields, strings.ilike(.name, "In-Reply-To")))
+      (
+        length(headers.references) > 0
+        or any(headers.hops, any(.fields, strings.ilike(.name, "In-Reply-To")))
+      )
+      // catch fake message threads nested within legit replies
+      and not (
+        any([body.html.display_text, body.plain.raw],
+            3 of (
+              strings.icontains(., "from:"),
+              strings.icontains(., "to:"),
+              strings.icontains(., "sent:"),
+              strings.icontains(., "date:"),
+              strings.icontains(., "cc:"),
+              strings.icontains(., "subject:")
+            )
+        )
+        and (
+          // newly registered sender domain
+          network.whois(sender.email.domain).days_old <= 30
+          // or suspiciously long HTML div id
+          or regex.icontains(body.html.raw, '<div\sid="(.{200,})".*>')
```

matching on "not a `"` repeating at least 200 times, ensures that `"` is not matched within the `.{200,}` logic. 
```suggestion
          or regex.icontains(body.html.raw, '<div\sid=\"[^\"]{200,}\".*>')
```